lib/msal-common/src/unifiedCache/utils/CacheHelper.ts

```diff
@@ -43,4 +45,86 @@ export class CacheHelper {
         });
         return Object.assign({}, ...keyValues);
     }
+
+    /**
+     *
+     * @param key
+     * @param homeAccountId
+     */
+    static matchHomeAccountId(key: string, homeAccountId: string): boolean {
+        return homeAccountId === key.split(Separators.CACHE_KEY_SEPARATOR)[CacheKeyPosition.HOME_ACCOUNT_ID];
+    }
+
+    /**
+     *
+     * @param key
+     * @param environment
+     */
+    static matchEnvironment(key: string, environment: string): boolean {
+        return environment === key.split(Separators.CACHE_KEY_SEPARATOR)[CacheKeyPosition.ENVIRONMENT];
+    }
+
+    /**
+     *
+     * @param key
+     * @param credentialType
+     * // TODO: Confirm equality for enum vs string here
+     */
+    static matchCredentialType(key: string, credentialType: string): boolean {
+        return credentialType.toLowerCase() === key.split(Separators.CACHE_KEY_SEPARATOR)[CacheKeyPosition.CREDENTIAL_TYPE].toString().toLowerCase();
+    }
+
+    /**
+     *
+     * @param key
+     * @param clientId
+     */
+    static matchClientId(key: string, clientId: string): boolean {
+        return clientId === key.split(Separators.CACHE_KEY_SEPARATOR)[CacheKeyPosition.CLIENT_ID];
+    }
+
+    /**
+     *
+     * @param key
+     * @param realm
+     */
+    static matchRealm(key: string, realm: string): boolean {
+        return realm === key.split(Separators.CACHE_KEY_SEPARATOR)[CacheKeyPosition.REALM];
+    }
+
+    /**
+     *
+     * @param key
+     * @param target
+     */
+    static matchTarget(key: string, target: string): boolean {
+        return CacheHelper.targetsIntersect(key.split(Separators.CACHE_KEY_SEPARATOR)[CacheKeyPosition.TARGET], target);
+    }
+
+    /**
+     * returns a boolean if the sets of scopes intersect (scopes are stored as "target" in cache)
+     * @param target
+     * @param credentialTarget
+     */
+    static targetsIntersect(credentialTarget: string, target: string): boolean {
+        const targetSet = new Set(target.split(" "));
```

Comment:
```suggestion
        const targetSet = ScopeSet.fromString(target);
        const credentialTargetSet = ScopeSet.fromString(credentialTarget);
        return targetSet.intersectingScopeSets(credentialTargetSet);
```

Reply:
As mentioned above, will move this to ScopeSet in a follow up PR.